### ISSUE TITLE
Substituted varkey not found in BoundedConstraintSet

### ISSUE THREAD

**Issue by pgkirsch** [hoburg/gpkit#905]
``` python
In [1]: from gpkit import Variable, Model

In [2]: from gpkit.constraints.bounded import BoundedConstraintSet

In [3]: x = Variable('x')

In [4]: y = Variable('y')

In [5]: m = Model(x, [x >= y], {'y':1})

In [6]: bm = Model(m.cost, BoundedConstraintSet(m))

In [7]: bm.solve()
Using solver 'cvxopt'
Solving for 1 variables.
Solving took 0.022 seconds.
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-7-14295aee770d> in <module>()
----> 1 bm.solve()

C:\Users\philippe.kirschen\Documents\Optimization\gpkit\gpkit\constraints\prog_factories.pyc in solvefn(self, solver, verbosity, skipswee
pfailures, *args, **kwargs)
     89         if "sweepvariables" in solution:
     90             solution["variables"].update(solution["sweepvariables"])
---> 91         self.process_solution(solution)
     92         self.solution = solution  # NOTE: SIDE EFFECTS
     93         return solution

C:\Users\philippe.kirschen\Documents\Optimization\gpkit\gpkit\constraints\set.pyc in process_solution(self, sol)
    285         for constraint in self:
    286             if hasattr(constraint, "process_solution"):
--> 287                 constraint.process_solution(sol)

C:\Users\philippe.kirschen\Documents\Optimization\gpkit\gpkit\constraints\bounded.pyc in process_solution(self, sol)
     86             if abs(lam_lt) >= 1e-7:  # arbitrary threshold
     87                 out["sensitive to lower bound"].append(varkey)
---> 88             value = mag(sol["freevariables"][varkey])
     89             distance_below = np.log(value/self.lowerbound)
     90             distance_above = np.log(self.upperbound/value)

C:\Users\philippe.kirschen\Documents\Optimization\gpkit\gpkit\keydict.pyc in __getitem__(self, key)
     97         if not keys:
     98             del self.keymap[key] # remove blank entry added due to defaultdict
---> 99             raise KeyError("%s was not found." % key)
    100         values = []
    101         for key in keys:

KeyError: 'y was not found.'

```

**Comment by bqpd** [comment]
Weird, it looks like you may have a different implementation of defaultdict? I can make a branch that fixes this, but for now could you try editing the indented block line 98 -100 to

```python
if key not in self.keymap:
    print key, type(key)
    print self.keymap
    raise  # remove this line to have everything probably work
else:
    del self.keymap[key] # remove blank entry added due to defaultdict
    raise KeyError("%s was not found." % key)
```

and tell me what that prints out?

**Comment by pgkirsch** [comment]
Had no effect, still got the same error and it didn't print anything out.

**Comment by pgkirsch** [comment]
I'm not sure why I would have a different implementation of anything, I'm pulling the latest master to use bounded.py

**Comment by 1ozturkbe** [comment]
Same here, and got the same error in gpkit-models/1682master. 
commit 5c9d286b263faf1d44ffa415543876c73bb4a330, within ~\gpkit-models\1682\gas_male\tailSizing.py

S_GasMALE has been very clearly substituted, but I get the following error. 
```python
In [1]: run tailSizing.py
Using solver 'mosek'
Solving for 28 variables.
Solving took 0.017 seconds.
-----------------------------------------------------------------------
----
KeyError                                  Traceback (most recent call l
ast)
C:\Users\Berk\Dropbox (MIT)\MIT Graduate School\16.82\code\gpkit-models
\1682\gas_male\tailSizing.py in <module>()
    196     #M.substitutions.update({'\\lambda_{htail}':0.8})
    197     M       = Model(M.cost, BCS(M))
--> 198     sol     = M.solve("mosek")
    199     varVals = sol['variables']
    200     print 'Tail downforce: ' + str(varVals['F_{boom}'])

c:\users\berk\dropbox (mit)\mit graduate school\code\gpkit\gpkit\constr
aints\prog_factories.pyc in solvefn(self, solver, verbosity, skipsweepf
ailures, *args, **kwargs)
     89         if "sweepvariables" in solution:
     90             solution["variables"].update(solution["sweepvariabl
es"])
---> 91         self.process_solution(solution)
     92         self.solution = solution  # NOTE: SIDE EFFECTS
     93         return solution

c:\users\berk\dropbox (mit)\mit graduate school\code\gpkit\gpkit\constr
aints\set.pyc in process_solution(self, sol)
    302         for constraint in self:
    303             if hasattr(constraint, "process_solution"):
--> 304                 constraint.process_solution(sol)

c:\users\berk\dropbox (mit)\mit graduate school\code\gpkit\gpkit\constr
aints\bounded.pyc in process_solution(self, sol)
     86             if abs(lam_lt) >= 1e-7:  # arbitrary threshold
     87                 out["sensitive to lower bound"].append(varkey)
---> 88             value = mag(sol["freevariables"][varkey])
     89             distance_below = np.log(value/self.lowerbound)
     90             distance_above = np.log(self.upperbound/value)

c:\users\berk\dropbox (mit)\mit graduate school\code\gpkit\gpkit\keydic
t.pyc in __getitem__(self, key)
     97         if not keys:
     98             del self.keymap[key] # remove blank entry added due
 to defaultdict
---> 99             raise KeyError("%s was not found." % key)
    100         values = []
    101         for key in keys:

KeyError: 'S_GasMALE was not found.'
```
This seems very similar (if not identical) to the error that Philippe has been experiencing. I am also using the master branch of gpkit.

**Comment by bqpd** [comment]
Still have not been able to reproduce this bug; @1ozturkbe's example runs on my computer on the current master fine. @1ozturkbe, could you confirm that the code you reference above does not run on the current master on your computer?

**Comment by 1ozturkbe** [comment]
So odd... I ran it on master and it worked fine. I will let you know if it comes up again, but I think we can mark it as resolved for now.

**Comment by whoburg** [comment]
This does not appear to be resolved, see ff07f98e4709096eea481a2605d73e44058dfa57 on #976

**Comment by bqpd** [comment]
That test raises a different error.
